### PR TITLE
RDDRONE-UAVCAN Added HRT & RTC timer support

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
**Describe problem solved by this pull request**
Added HRT & RTC timer support for RDDRONE-UAVCAN

**Describe your solution**
Replaced Kinetis stubs with S32K FTM driver
Enabled S32K RTC driver

**Test data / coverage**
Tested on RDDRONE-UAVCAN board checking whether the top command is working correctly.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Nice, thanks!

File: platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- *   Copyright (c) 2016-2017 PX4 Development Team. All rights reserved.
+ *   Copyright (c) 2016-2019 PX4 Development Team. All rights reserved.
```
Comment (davids5) [inline platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]:
```suggestion
 *   Copyright (c) 2016-2020 PX4 Development Team. All rights reserved.
```

File: platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c [hunk 2]
```diff
@@ -192,27 +140,106 @@ hrt_absolute_time(void)
...
+#define rC1SC       REG(0x14)
+#define rC1V        REG(0x18)
+#define rSTATUS     REG(0x50)
+#define rCOMBINE    REG(0x64)
+#define rPOL        REG(0x70)
+#define rFILTER     REG(0x78)
+#define rQDCTRL     REG(0x80)
+#define rCONF       REG(0x84)
+
+/* SC Bit Fields */
```
Comment (davids5) [inline platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: Can we upstream the header files to nuttx and bring them in by inclusion?

File: platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c [hunk 3]
```diff
@@ -330,50 +362,37 @@ static void	hrt_ppm_decode(uint32_t status);
...
-	rSC = TPM_SC_TOF;
+	rSC = FTM_SC_TOF(1);
 
 	rCNT = 0;
 	rMOD = HRT_COUNTER_PERIOD - 1;
 
-	rSTATUS   = (TPM_STATUS_TOF | STATUS_HRT | STATUS_PPM);
-	rCNSC_HRT = (TPM_CnSC_CHF | TPM_CnSC_CHIE | TPM_CnSC_MSA);
-	rCNSC_PPM = (TPM_CnSC_CHF | CNSC_PPM);
+	//rSTATUS   = (STATUS_HRT | STATUS_PPM);
```
Comment (davids5) [inline platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: delete

File: platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c (hunk 2 above)
Comment (PetervdPerk-NXP) [reply platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: Yeah this was a work around for the header files, but I can make a FTM register header file and upstream to NuttX.

Comment (davids5) [reply platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: OK if you do that I can backport it ASAP

Comment (PetervdPerk-NXP) [reply platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: Done see https://github.com/apache/incubator-nuttx/pull/614

Comment (davids5) [reply platforms/nuttx/src/px4/nxp/s32k1xx/hrt/hrt.c]: Done https://github.com/PX4/NuttX/pull/95 - @dagar Please appove/merge

**Comment by davids5** [comment]
@PetervdPerk-NXP - please run `make format`

**Comment by PetervdPerk-NXP** [comment]
> Might be a different astyle version? I have 2.06.

I'm running Ubuntu 18.04 on which astyle:amd64/bionic 3.1-1ubuntu2 uptodate was installed.

```
hovergames@hovergames:~/nuttx/PX4-Firmware$ astyle --version
Artistic Style Version 3.1
hovergames@hovergames:~/nuttx/PX4-Firmware$ make format
Formatting with astyle 

diff --git a/platforms/common/include/px4_platform_common/i2c_spi_buses.h b/platforms/common/include/px4_platform_common/i2c_spi_buses.h.pretty
index 106dfec0c9..f2ba7b55c2 100644
--- a/platforms/common/include/px4_platform_common/i2c_spi_buses.h
+++ b/platforms/common/include/px4_platform_common/i2c_spi_buses.h.pretty
@@ -235,11 +235,11 @@ protected:
 
 	virtual ~I2CSPIDriver() = default;
 
-	void Run() final {
+	void Run() final
+	{
 		static_cast<T *>(this)->RunImpl();
 
-		if (should_exit())
-		{
+		if (should_exit()) {
 			exit_and_cleanup();
 		}
 	}

Formatted  platforms/common/include/px4_platform_common/i2c_spi_buses.h
Format checks passed
```

Downgrading to 2.06 solves the issue
```
hovergames@hovergames:~/nuttx/PX4-Firmware$ astyle --version
Artistic Style Version 2.06
hovergames@hovergames:~/nuttx/PX4-Firmware$ make format
Formatting with astyle 

diff --git a/platforms/common/include/px4_platform_common/i2c_spi_buses.h b/platforms/common/include/px4_platform_common/i2c_spi_buses.h.pretty
index f2ba7b55c2..106dfec0c9 100644
--- a/platforms/common/include/px4_platform_common/i2c_spi_buses.h
+++ b/platforms/common/include/px4_platform_common/i2c_spi_buses.h.pretty
@@ -235,11 +235,11 @@ protected:
 
 	virtual ~I2CSPIDriver() = default;
 
-	void Run() final
-	{
+	void Run() final {
 		static_cast<T *>(this)->RunImpl();
 
-		if (should_exit()) {
+		if (should_exit())
+		{
 			exit_and_cleanup();
 		}
 	}

Formatted  platforms/common/include/px4_platform_common/i2c_spi_buses.h
Format checks passed
```

**Comment by davids5** [comment]
@bkueng , @PetervdPerk-NXP 
LOL mine is "Artistic Style Version 3.1" we have to use the same version that CI uses. @bkueng do you have control over CI?

**Comment by bkueng** [comment]
CI is using 2.06 as well: https://github.com/PX4/containers/blob/master/docker/Dockerfile_base-bionic#L73

**Comment by davids5** [comment]
> CI is using 2.06 as well: https://github.com/PX4/containers/blob/master/docker/Dockerfile_base-bionic#L73

I think following the dev guide and installing using the scripts is the root cause here.

@dagar Could it be our tools are out of sync?  

https://raw.githubusercontent.com/PX4/Devguide/v1.9.0/build_scripts/ubuntu_sim_common_deps.sh
```
# Common dependencies
echo "Installing common dependencies"
sudo apt-get update -y
sudo apt-get install git zip qtcreator cmake build-essential genromfs ninja-build exiftool astyle -y
# make sure xxd is installed, dedicated xxd package since Ubuntu 18.04 but was squashed into vim-common before
which xxd || sudo apt install xxd -y || sudo apt-get install vim-common --no-install-recommends -y
# Required python packages
sudo apt-get install python-argparse python-empy python-toml python-numpy python-dev python-pip -y
```

```
aptitude search astyle -F "%c %p %d %V"
i astyle   Source code indenter for C, C++, Objective-C, C#, and Java 3.1-1ubuntu2  
...
```

**Pull request merged**